app/controllers/notes_controller.rb

```diff
@@ -55,10 +55,56 @@ def destroy
       track_destroyed(@note)
       redirect_to node_path(@node), notice: 'Note deleted'
     else
-      redirect_to node_note_path(@node, @note), alert: 'Could not delete node'
+      redirect_to node_note_path(@node, @note), alert: 'Could not delete note'
     end
   end
 
+  def multi_destroy
+    @notes = Note.where(
+      'node_id = :node AND id in (:ids)',
+      node: @node,
+      ids: params[:ids]
+    )
+
+    if @notes.any?
+      @job_logger = Log.new(uid: (Log.maximum(:uid) || 0) + 1)
+
+      if @notes.count > Note::MAX_DELETED_INLINE
+        @job_logger.write 'Enqueueing destroy job to start in the background.'
+        job = DestroyJob.perform_later(
+          items: @notes.to_a,
+          author_email: current_user.email,
+          uid: @job_logger.uid
+        )
+        @job_logger.write "Job id is #{ job.job_id }."
+
+      elsif @notes.count > 0
+        @job_logger.write 'Performing destroy job inline.'
+        DestroyJob.perform_now(
+          items: @notes.to_a,
+          author_email: current_user.email,
+          uid: @job_logger.uid
+        )
+      end
+    end
+
+    respond_to do |format|
+      format.html do
+        redirect_to node_path(@node, tab: 'notes-tab'),
+                    notice: 'Deleting notes...'
+      end
+      format.json
+    end
+  end
+
+  def multi_destroy_status
+    @logs = Log.where(
+      'uid = ? and id > ?',
+      params[:item_id].to_i, params[:after].to_i
+    )
+    @destroying = !(@logs.last.text == 'Worker process completed.') if @logs.any?
```

Comment:
`@logs.last.text != 'Worker process completed.'` is more concise

Reply:
👍